tutorial/creating_packages/package_information/conanfile.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+from conan.tools.scm import Git
+from conan.tools.build import check_min_cppstd
+
+
+class helloRecipe(ConanFile):
+    name = "hello"
+    version = "1.0"
+
+    # Optional metadata
+    license = "<Put the package license here>"
+    author = "<Put your name here> <And your email here>"
+    url = "<Package recipe repository url here, for issues about the package>"
+    description = "<Description of hello package here>"
+    topics = ("<Put some tag here>", "<here>", "<and here>")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_fmt": [True, False]}
+
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_fmt": True}
+
+    generators = "CMakeDeps"
+
+    def validate(self):
+        if self.info.options.with_fmt:
+            check_min_cppstd(self, "11")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        git = Git(self)
+        git.clone(url="https://github.com/conan-io/libhello.git", target=".")
+        # Please, be aware that using the head of the branch instead of an inmutable tag
+        # or commit is not a good practice in general
+        git.checkout("package_info")
+
+    def requirements(self):
+        if self.options.with_fmt:
+            self.requires("fmt/8.1.1")
+        self.test_requires("gtest/1.11.0")
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.options.with_fmt:
+            tc.variables["WITH_FMT"] = True
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+        # this check is not needed if using CTest instead of gtest
+        # in that case just call to cmake.test() and it will be skipped
+        # if tools.build:skip_test=True
+        if not self.conf.get("tools.build:skip_test", default=False):
```

Comment:
I think this deserves a feature for 2.X, running a ``test()`` method to run package unit-tests, just after the build(), and inhibited by ``skip_test``. It would be simple, clean, and useful.